esteid/bdoc2/README.md

```diff
@@ -0,0 +1,282 @@
+
+The other, most important, fields are the `Reference` ones. And they are quite different in purpose and formation.
+
+The first `Reference` field is about the signed document and as such, has an `URI` attribute of the document's file name.
+Its child `DigestValue` element is the SHA256 hash of the document, it is, incidentally, the very hash that is sent to the SmartID API for signing.
+
+The second `Reference` is an interesting beast, as it is built on the basis of some fields defined later in the [SignedProperties](#SignedProperties) section.
+Its child `DigestValue` is calculated as a SHA256 hash of the canonicalized XML output of the  `SignedProperties` tag, after that one is formed:
+The `URI` attribute of this `Reference` tag is the `#`-prefixed `Id` attribute of the `SignedProperties` tag.
+
+```python
+import base64
+import hashlib
+from lxml import etree
+
+buf = etree.tostring(el, method='c14n', exclusive=True or False)  # NOTE below
+digest_value = base64.b64encode(hashlib.sha256(buf).digest())
+```
+(Assuming the `el` here to be the XML `<SignedProperties>` element)
+
+#### Canonicalization
+
+The `exclusive` kwarg controls whether the namespace declarations of ancestor tags should be included in the resulting canonical representation, or _excluded_.
+Whether to use `exclusive=True` depends on the canonicalization tag's `Algorithm` attribute: 
+* `http://www.w3.org/2001/10/xml-exc-c14n#`, uses `exclusive=True`,  
+* the two others, the required `http://www.w3.org/TR/2001/REC-xml-c14n-20010315`, or `http://www.w3.org/2006/12/xml-c14n11`, are not exclusive.
+
+The aforementioned `<ds:CanonicalizationMethod>` tag controls the c14n of the `SignedInfo` node before feeding its digest to the signature service. 
+The c14n of `SignedProperties` prior to getting its digest is determined by the `ds:Transform` tag within this `ds:Reference` node. 
+If it's not present, then the default, ie. not exclusive, c14n is used.  
+
+### KeyInfo
+
+This section contains just the certificate value as gotten from the SmartID API response's `cert.value`:
+```xml
+<ds:KeyInfo Id="S0-KeyInfo">
+    <ds:X509Data>
+        <ds:X509Certificate>MIIGJDCCBAygAwIBAgIQBNsLtTIpnmNbbE4+laSLaTANBgkqhkiG9w0BAQsFADBr...</ds:X509Certificate>
+    </ds:X509Data>
+</ds:KeyInfo>
+```
+More details about the certificate in the [SigningCertificate](#SigningCertificate) subsection.
+ 
+### SignedProperties
+
+The XML section of `SignedProperties` consists of, at least :question:, the `SigningTime`, `SigningCertificate` and `SignaturePolicyIdentifer` elements.
+
+#### SigningTime
+
+Is a timestamp in ISO 8601 format.
+
+:question: Is it enough to put a date of the SmartID sign request here? No obvious way to get the date from the SmartID API.
+
+#### SignaturePolicyIdentifier 
+
+This appears to be a static XML chunk referencing the BDOC 2.1 Specifications document:
```

Comment:
Spec example for `ds:DigestValue` in `xades:SigPolicyHash` contains this:

```
<ds:DigestValue>*** HERE IS THE HASH VALUE OF THE CURRENT
DOCUMENT IN BASE64 ENCODING***</ds:DigestValue>
```

Which looks like not-static to me. The comment here confused me so I'll update the wording.

Reply:
It's kinda static since this document's hash value will hardly change, and other fields too, but probably this clause is worth including as is